daft/api_annotations.py

```diff
@@ -43,7 +52,11 @@ def PublicAPI(func: Callable[P, T]) -> Callable[P, T]:
     def _wrap(*args: P.args, **kwargs: P.kwargs) -> T:
         __tracebackhide__ = True
         type_check_function(func, *args, **kwargs)
-        return func(*args, **kwargs)
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            e = e.with_traceback(e.__traceback__.tb_next if e.__traceback__ else None)
```

Comment:
This will also affect other tracebacks by hiding this file. I don't think this function needs the special case for `except UDFException`, but can still add it